components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java

```diff
@@ -30,36 +31,40 @@
     /**
      * This API is used to create the given secret.
      *
-     * @param secret The {@link Secret}.
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @param secret         The {@link Secret}.
      * @return Returns {@link Secret} created.
      * @throws SecretManagementException Secret management exception.
      */
-    Secret addSecret(Secret secret) throws SecretManagementException;
+    Secret addSecret(String secretTypeName, Secret secret) throws SecretManagementException;
 
     /**
      * This API is used to retrieve the given secret.
      *
-     * @param secretName Name of the {@link Secret}.
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @param secretName     Name of the {@link Secret}.
      * @return Returns {@link Secret} requested.
      * @throws SecretManagementException Secret management exception.
      */
-    Secret getSecret(String secretName) throws SecretManagementException;
+    Secret getSecret(String secretTypeName, String secretName) throws SecretManagementException;
 
     /**
      * Get all the secrets of the current tenant.
      *
+     * @param secretTypeName Name of the {@link SecretType}.
      * @return {@link Secrets} object with all the tenant secrets.
      * @throws SecretManagementException Secret Management Exception.
      */
-    Secrets getSecrets() throws SecretManagementException;
+    Secrets getSecrets(String secretTypeName) throws SecretManagementException;
 
     /**
      * This API is used to delete the given secret.
      *
-     * @param secretName Request to delete the {@link Secret}.
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @param secretName     Request to delete the {@link Secret}.
```

Comment:
Shouldn't this be `Name of the {@link Secret}.`

Reply:
Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)